### ISSUE TITLE
In director-v0: Better error handling when save-data request to dynamic services fails.

### ISSUE THREAD

**Issue by mrnicegyu11** [ITISFoundation/osparc-simcore#2800]
If a study with dynamic services (we are primarily concerned with jupyter notebooks here) is closed, the director-v0 instructs the dynamic service to shut down gracefully. This triggers a hook in the jupyter-notebook that saves the data to S3.

If this request, from the director-v0 to the jupyter, fails for some reason (timeout or non-200 return code), the following happens:
- The director-v0 logs `service %s could not be contacted, state not saved` (`producer.py` lines 1036 and further)
- The service is removed, i.e. the container is stopped and removed from the machine

This implies definite data-loss if the REST POST request in `producer.py` line 1041 fails.

More graceful error handling may include:
- Trying again a few times to trigger the POST request
- Keep the container (just stop it) if the save-request falls through. Then, manual data recovery (logging into the container and zipping the data) would be possible. Then however, undeleted containers would bloat the machine in the long run. 

On Friday, some user data was impacted by this.

**Comment by mrnicegyu11** [comment]
Another idea from the POs to remedy this: `perhaps we should add a save button that a user can use to save before leaving. potentially even multiple times.`

**Comment by KZzizzle** [comment]
Just also want to make clear that the source of Taylor's trouble was that "At a certain point, I started experiencing connectivity issues in one of my jupyter NBs, so I closed and re-opened the study". During the second incident, Taylor experienced server connections again and did NOT close and reopen the study. Instead, the jupyterLab containers restarted themselves and after that, the data were lost. It's not clear that even if saving had worked perfectly that the data would have been preserved after the connectivity issues.
![image](https://user-images.githubusercontent.com/28002886/152753633-60e561be-f790-4a90-8071-ab5277c83b06.png)
